File: web-app/src/app/core/models/usuario.ts
export interface LoginRequest {
  correoElectronico: string;
  contrasena: string;
}

export interface LoginResponse {
  token: string;
  tipo: string;
  id: string;
  nombreUsuario: string;
  correoElectronico: string;
  nombres: string;
  apellidos: string;
  rol: string;
  rolId: string;
}

export interface UsuarioResponse {
  id: string;
  nombreUsuario: string;
  correoElectronico: string;
  nombres: string;
  apellidos: string;
  rolId: string;
  rolNombre: string;
  estado: string;
  ultimoAcceso: string;
  fechaCreacion: string;
  fechaActualizacion: string;
}

export interface UsuarioRequest {
  nombreUsuario: string;
  correoElectronico: string;
  contrasena: string;
  nombres: string;
  apellidos: string;
  rolId: string;
}

export interface UsuarioUpdateRequest {
  nombres?: string;
  apellidos?: string;
  correoElectronico?: string;
  estado?: string;
}

export interface CambiarPasswordRequest {
  contrasenaActual: string;
  nuevaContrasena: string;
}

export interface ForgotPasswordRequest {
  correoElectronico: string;
}

export interface ResetPasswordRequest {
  token: string;
  newPassword: string;
}

File: api-rest/hotel_cervera_api/src/main/java/com/hotel/cervera/hotel_cervera_api/model/Usuario.java
package com.hotel.cervera.hotel_cervera_api.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.OffsetDateTime;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
@Table(name = "usuarios")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "nombre_usuario", nullable = false, unique = true, length = 50)
    private String nombreUsuario;

    @Column(name = "correo_electronico", nullable = false, unique = true, length = 100)
    private String correoElectronico;

    @JsonIgnore
    @Column(name = "contrasena_hash", nullable = false, length = 255)
    private String contrasenaHash;

    @Column(name = "nombres", nullable = false, length = 50)
    private String nombres;

    @Column(name = "apellidos", nullable = false, length = 50)
    private String apellidos;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "rol_id", nullable = false)
    private Rol rol;

    @Column(name = "estado", nullable = false, length = 20)
    private String estado;

    @Column(name = "ultimo_acceso")
    private OffsetDateTime ultimoAcceso;

    @Column(name = "intentos_fallidos")
    private Integer intentosFallidos;

    @Column(name = "bloqueado_hasta")
    private OffsetDateTime bloqueadoHasta;

    @CreationTimestamp
    @Column(name = "fecha_creacion", updatable = false)
    private OffsetDateTime fechaCreacion;

    @UpdateTimestamp
    @Column(name = "fecha_actualizacion")
    private OffsetDateTime fechaActualizacion;

    @Column(name = "fecha_eliminacion")
    private OffsetDateTime fechaEliminacion;
}
